src/wip/streaming-dca.rain

```diff
@@ -0,0 +1,149 @@
+          bounty-unit-time: 60
+
+          twap-io-fee: '[uniswap-v3-fee-low]'
+
+        scenarios:
+         buy:
+          bindings:
+           min-ratio: 1
+           calculate-io-ratio: '''calculate-io-ratio-buy'
+         sell:
+          bindings:
+           min-ratio: 100
+           calculate-io-ratio: '''calculate-io-ratio-sell'
+          
+                                    
+---
+#distribution-token !The distribution token, the token that is bought and sold against the stable token.
+#reserve-token !The token that will be used to compare against the stable token to calculate the TWAP for dollar equivalent conversions. 
+
+#twap-io-fee !Twap fee for the distribution token pool, paired with either stable token. 
+#calculate-io-ratio !Binding to calculate io-ratio for the order.
+#min-ratio !Minimum input ouput ratio for the order.
+
+#bounty-min !Minimum bounty to offer for each trade in USD.
+#bounty-unit-increase !Amount to increase bounty per unit time.
+#bounty-unit-time !Unit of time to increase the bounty.
+#bounty-max !Maximum bounty that can be offered. 
+
+#per-second-rate !The rate of stable denominated distributed tokens distributed per second.
+#strat-strat-time !Strategy start timestamp
+
+#uniswap-words !The subparser for the Uniswap words
+#orderbook-subparser !The subparser for the Orderbook words
+#init-key "init-key"
+#last-time-key "last-time-key"
+
+#bounty-auction
+  time-since-cooldown: ,
+  bounty: min(
+    linear-growth(
+      bounty-min
+      div(bounty-unit-increase bounty-unit-time)
+      time-since-cooldown
+    )
+    bounty-max
+  );
+
+#get-last-time
+ is-initialized: get(hash(order-hash() init-key)),
+ current-time: block-timestamp(),
+ last-time: if(
+    is-initialized
+    get(hash(order-hash() last-time-key))
+    strat-strat-time
+  );
+
+#set-last-time
+ :set(hash(order-hash() init-key) 1),
+ :set(hash(order-hash() last-time-key) block-timestamp());
+
+#calculate-io-ratio-buy
+  usd-amount 
+  bounty-amount: ,
+  output-amount: uniswap-v3-quote-exact-input(reserve-token distribution-token usd-amount twap-io-fee),
+  _: usd-amount,
+  io-ratio: div(output-amount add(usd-amount bounty-amount)),
+  :ensure(
+    greater-than-or-equal-to(io-ratio min-ratio)
+    "min ratio"
+  );
+
+#calculate-io-ratio-sell
+  usd-amount 
+  bounty-amount: ,
+  output-amount: uniswap-v3-quote-exact-output(distribution-token reserve-token usd-amount twap-io-fee),
+  io-ratio: div(saturating-sub(usd-amount bounty-amount) output-amount),
+  :ensure(
+    greater-than-or-equal-to(io-ratio min-ratio)
+    "min ratio"
+  );
+
+#calculate-io
+  using-words-from uniswap-words orderbook-subparser
+  
+  current-time last-time: call<'get-last-time>(), 
+  elapsed-time: saturating-sub(current-time last-time),
+  usd-amount: mul(per-second-rate elapsed-time),
+  max-output-amount io-ratio: call<'calculate-io-ratio>(usd-amount call<'bounty-auction>(elapsed-time));
+  
+#handle-io
+ :call<'set-last-time>();
```

Comment:
this doesn't need to happen in handle io because it's not contingent on the counterparty

Reply:
All addressed in the recent push